hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreCompactor.java

```diff
@@ -229,7 +236,7 @@ private void doCompaction() {
   }
 
   /**----------------------------------------------------------------------
-   * The copy-compaction is the creation of the ImmutableSegment (from the relevant type)
+   * The copy-compaction is the creation of the ImmutableSegment (from the relevant action)
    * based on the Compactor Iterator. The new ImmutableSegment is returned.
    */
   private ImmutableSegment compact(int numOfCells) throws IOException {
```

Comment:
I think it would be ok to pass the action (flatten/merge/compaction) as parameter, instead of getting it from an attribute.
Is it the case that we always know the action from the context in which we trigger the compact method?